Zend/zend_alloc.c

```diff
@@ -2923,6 +2983,7 @@ static void alloc_globals_ctor(zend_alloc_globals *alloc_globals)
 #ifdef ZTS
 static void alloc_globals_dtor(zend_alloc_globals *alloc_globals)
 {
+	zend_mm_observers_shutdown(alloc_globals->mm_heap);
```

Comment:
If `zend_mm_shutdown` removes the `full == false` check this line can be removed.